javascript/src/openai/responses.ts

```
@@ -0,0 +1,418 @@
+        }
+        case "tool": {
+          // Tool results in Responses API need to be converted to function call outputs
+          // For now, we'll represent them as assistant messages
+          for (const toolResult of message.content) {
+            responseMessages.push({
+              role: "assistant",
+              content: [
+                {
+                  type: "input_text",
+                  text: `Tool result for ${toolResult.toolCallId}: ${JSON.stringify(toolResult.result)}`,
+                },
+              ],
+              type: "message",
+            });
+          }
+          break;
+        }
+      }
+    }
+
+    return responseMessages;
+  }
+
+  /**
+   * Map OpenAI response events to content deltas
+   */
+  private mapEventToContentDeltas(
+    event: OpenAI.Responses.ResponseStreamEvent,
+  ): ContentDelta[] {
+    const contentDeltas: ContentDelta[] = [];
+
+    switch (event.type) {
+      case "response.reasoning_summary_text.delta": {
+        const part: ReasoningPartDelta = {
+          type: "reasoning",
+          reasoning: event.delta,
+          summary: true,
+        };
+        contentDeltas.push({
+          index: event.output_index || 0,
+          part,
+        });
+        break;
+      }
+      case "response.reasoning.delta": {
+        const part: ReasoningPartDelta = {
+          type: "reasoning",
+          reasoning: event.delta as string,
+          summary: false,
+        };
+        contentDeltas.push({
+          index: event.output_index || 0,
+          part,
+        });
+        break;
+      }
+      case "response.reasoning.done": {
+        // When reasoning is done, we get the complete text
+        // The signature should be available in the final response
+        const part: ReasoningPartDelta = {
+          type: "reasoning",
+          reasoning: event.text,
+          summary: false,
+        };
+        contentDeltas.push({
+          index: event.output_index || 0,
```

remove all the `|| 0`. If the value is not defined, we must throw an error. Instead this silently ignore the error and overwrite things or push mixed parts into the same part slot.